### ISSUE TITLE
3.0.191 - Fatal Error: Allowed memory size of 536870912 bytes exhausted

### ISSUE THREAD

**Issue by jlahijani** [processwire/processwire-issues#1495]
I updated to 3.0.191 on one of my sites, and when I try to access the page tree in the admin, I get this:

```
Fatal Error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 16384 bytes)
Line 166 of /Users/jlahijani/Sites/processwire-dev/wire/core/WireDatabasePDOStatement.php
```

Downgrading to 3.0.190 resolved the issue.  This issue is only happening on my of my sites (the others are fine).

I can access other parts of the admin, but going to the tree is where this issue occurs.

**Comment by ryancramerdesign** [comment]
@jlahijani There was an issue with the first committed version of 3.0.191 that caused issues with creating new pages, but maybe it could be related to the issue you are seeing. It was fixed in [this commit](https://github.com/processwire/processwire/commit/3c983f9b51d05160e773fd9ce5615a067cd87418) ... do you already have that, or is the issue you are seeing unrelated?

**Comment by jlahijani** [comment]
@ryancramerdesign I already had that commit.  I tested again with the very latest as of this moment and I'm still having the same issue.  I'm not sure how to troubleshoot this issue specifically, but if you have any (advanced) tips on how to catch where this hangup is occurring I can look into it.

**Comment by ryancramerdesign** [comment]
@jlahijani That error appears to be happening on the `parent::execute($input_parameters)` call (line 166 in WireDatabasePDOStatement.php), which means it's occurring inside PHP's PDOStatement class. We can't debug inside that class since it's part of the compiled PHP source.  But you could add the following code before that parent call so that we could see what's in it and might give us clues about what's causing the PDOStatement's execute() method to run out of memory. You'll get a lot of output here so it would just be the last bit of it before the error that would be of interest. 

With Tracy:
~~~~~
static $n = 0; $n++;
bd($this->queryString, "$n. SQL");
bd($input_parameters, "$n. Params");
~~~~~

Without Tracy:
~~~~~
static $n = 0; $n++;
echo "\n<pre>";
echo "\n$n. SQL: " . htmlspecialchars($this->queryString);
echo "\n$n. Params: " . htmlspecialchars(print_r($input_parameters, true));
echo "\n</pre>\n";
~~~~~

**Comment by jlahijani** [comment]
@ryan Using the non-Tracy method, I outputted the contents to one of the PW log files instead of directly in the browser since it was creating login issues.  Here it is... it keeps repeating these set of queries over and over again:

```
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22277. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22277. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22278. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22278. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22279. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22279. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22280. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22280. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22281. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22281. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22282. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22282. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22283. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22283. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22284. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22284. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22285. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22285. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22286. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22286. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22287. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22287. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22288. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22288. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22289. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22289. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22290. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22290. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22291. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22291. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22292. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22292. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22293. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22293. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22294. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22294. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22295. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22295. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22296. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22296. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22297. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22297. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22298. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22298. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22299. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22299. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22300. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22300. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22301. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22301. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22302. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22302. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22303. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22303. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22304. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22304. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22305. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22305. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22306. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22306. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22307. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22307. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22308. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22308. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22309. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22309. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22310. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22310. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22311. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22311. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22312. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22312. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22313. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22313. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22314. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22314. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22315. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22315. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22316. SQL: SELECT false AS isLoaded, pages.templates_id AS templates_id, pages.*, pages_sortfields.sortfield,  (SELECT COUNT(*) FROM pages AS children WHERE children.parent_id=pages.id) AS numChildren,field_title.data AS `title__data`   FROM `pages`   LEFT JOIN pages_sortfields ON pages_sortfields.pages_id=pages.id  LEFT JOIN field_title ON field_title.pages_id=pages.id  WHERE pages.id=:id  AND pages.templates_id=:templates_id GROUP BY pages.id  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22316. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22317. SQL: SELECT SQL_CALC_FOUND_ROWS pages.id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22317. Params:  (WireLog)
2022-01-01 18:09:00	jlahijani	https://www.mysite.com/admin/page/	22318. SQL: SELECT pages.id,pages.parent_id,pages.templates_id   FROM `pages`   WHERE (pages.templates_id=:i0X) GROUP BY pages.id  LIMIT 0,1  (WireLog)
```

**Comment by ryancramerdesign** [comment]
@jlahijani Thanks for trying that and reporting back. So far it's still a mystery. You mentioned this is just happening on one installation, so maybe we need to look at what's different about that installation. What modules do you have installed (Modules > Site) and any custom hooks in /site/*.php files or /site/templates/admin.php? Is it only the admin page list that produces this issue, or do you observe it elsewhere on this installation? Thanks.

**Comment by jlahijani** [comment]
@ryancramerdesign This site dates back to 2014.  In that time, using certain reserved words for fields like `user` or `users` was allowed, but as of the last couple years, a change was made where they are now reserved.  I've been aware of this but so far it hasn't bitten me.  I don't know if that's the issue (and I don't want you to waste your time on it), but something tells me that it might be, even though I tried a bunch of things without success.  I will look into this further some other day.  For now, that site will be locked on 3.0.190.

**Comment by ryancramerdesign** [comment]
@jlahijani If you'd like me to have a closer look feel free to email or PM me the DB export file and I'm happy to test with it here to see if I can spot what the issue is.

**Comment by jlahijani** [comment]
@ryancramerdesign I figured it out what it was and fixed it.  I got to the point where I disabled nearly all modules, then started to delete a bunch of pages and saw where it was hanging.  I think it's a bug in 3.0.191 as I was able to replicate the issue.  It has nothing to do with the reserved words as I mentioned previously.

The issue has to do with a template that has itself set as "Allowed templates for parents".  I was able to replicate it on a totally separate installation.  Here's how to test:

- create a template called 'x'
- create a page under home called 'x' using the 'x' template; publish it
- edit the 'x' template again and set "Allowed templates for parents" to 'x'
- go to page tree; it should stall / run out of memory and never load

Not sure why it's happening there, as I'm sure I've set a template to allow itself as a parent before.  Weird.
